src/utils/scroll-lock.js

```diff
@@ -73,23 +74,30 @@ function advancedUnlock(targetElement) {
   document.removeEventListener('touchmove', preventDefault);
 }
 
+const isVerticalScroll = ({ scrollHeight, scrollWidth }) => scrollHeight > scrollWidth;
+
```

Comment:
In the example you give, I don't think that would happen because in the case of a code block, the `scrollWidth` would be greater than the `scrollHeight`.

A narrow viewport wouldn't affect the `scrollWidth` since it's a measurement of the width of an element's content, including content not visible on the screen due to overflow. [Source](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollWidth).